lib/text/listener.js

```diff
@@ -83,6 +83,82 @@ class Importer extends SHRParserListener {
     }
   }
 
+    enterValuesetDefsDoc(ctx) {
+        const ns = ctx.valuesetDefsHeader().namespace().getText();
+        this._currentNs = ns;
+        if (typeof this._nsMap[ns] == 'undefined') {
+            this._nsMap[ns] = new Namespace(ns);
+        }
+    }
+
+    exitValuesetDefsDoc(ctx) {
+        // clear current namespace and uses namespaces
+        this._currentNs = '';
+        this._usesNs = [];
+    }
+
+    enterValuesetDef(ctx) {
+        const name = ctx.valuesetHeader().simpleName().getText();
+        const id = new Identifier(this._currentNs, name);
+        const valueset = new Valueset(id);
+
+        var path = this._preprocessedData.resolvePath('default', this._currentNs);
+
+        valueset.url = path.url + name;
+        valueset.name = name;
+
+        this._currentDef = valueset;
+    }
+
+    exitValuesetDef(ctx) {
+      this.pushCurrentDefinition();
```

Comment:
You'll probably need a separate `this.pushCurrentDefinitionToVSNamespace()` since you have to keep the namespaces separated.  (ugh.  I know).